src/aggregator/integration/one_client_pass_through_test.go

```diff
@@ -0,0 +1,147 @@
+// +build integration
+
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package integration
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3/src/cluster/placement"
+	maggregation "github.com/m3db/m3/src/metrics/aggregation"
+	"github.com/m3db/m3/src/metrics/metadata"
+	"github.com/m3db/m3/src/metrics/policy"
+	"github.com/m3db/m3/src/x/clock"
+	xtime "github.com/m3db/m3/src/x/time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestOneClientPassThroughMetrics(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+
+	serverOpts := newTestServerOptions()
+
+	// Clock setup.
+	var lock sync.RWMutex
+	now := time.Now().Truncate(time.Hour)
+	getNowFn := func() time.Time {
+		lock.RLock()
+		t := now
+		lock.RUnlock()
+		return t
+	}
+	setNowFn := func(t time.Time) {
+		lock.Lock()
+		now = t
+		lock.Unlock()
+	}
+	clockOpts := clock.NewOptions().SetNowFn(getNowFn)
+	serverOpts = serverOpts.SetClockOptions(clockOpts)
+
+	// Placement setup.
+	numShards := 1024
+	cfg := placementInstanceConfig{
+		instanceID:          serverOpts.InstanceID(),
+		shardSetID:          serverOpts.ShardSetID(),
+		shardStartInclusive: 0,
+		shardEndExclusive:   uint32(numShards),
+	}
+	instance := cfg.newPlacementInstance()
+	placement := newPlacement(numShards, []placement.Instance{instance})
+	placementKey := serverOpts.PlacementKVKey()
+	placementStore := serverOpts.KVStore()
+	require.NoError(t, setPlacement(placementKey, placementStore, placement))
+
+	// Create server.
+	testServer := newTestServerSetup(t, serverOpts)
+	defer testServer.close()
+
+	// Start the server.
+	log := testServer.aggregatorOpts.InstrumentOptions().Logger()
+	log.Info("test one client sending multiple types of timed metrics")
```

Comment:
curious about log messages within tests are these absolutely needed? Perhaps it would be best to create a series of test cases with names to denote which test is running?